src/agent/rustjail/src/mount.rs

```diff
@@ -1010,18 +1011,33 @@ lazy_static! {
     };
 }
 
+fn permissions_from_path(path: &Path) -> Result<u32> {
+    match fs::metadata(path) {
+        Ok(metadata) => {
+            let mut mode = metadata.permissions().mode();
+            if metadata.is_dir() {
+                // If it’s a directory, ensure “executable” bits are set.
```

Comment:
I don't understand the rationale... when called from `mknod_dev()`, we want to check the perms of a device node in the guest, not a directory.

Reply:
I may be overthinking this, I thought we may use this function somewhere else e.g. to check other guest based permissions. I can drop the is_dir() it is more concise then.

Reply:
Overthinking is bad :stuck_out_tongue_winking_eye:  Thanks @zvonkok !